drivers/sensor/bmi270/bmi270.c

```diff
@@ -0,0 +1,699 @@
+						   BMI270_GYR_FILT_NOISE,
+						   BMI270_GYR_FILT_NOISE_PERF);
+		} else {
+			gyr_conf = BMI270_SET_BITS(gyr_conf,
+						   BMI270_GYR_FILT,
+						   BMI270_GYR_FILT_PWR_OPT);
+			gyr_conf = BMI270_SET_BITS(gyr_conf,
+						   BMI270_GYR_FILT_NOISE,
+						   BMI270_GYR_FILT_NOISE_PWR);
+		}
+
+		dev->gyr_odr = odr_bits;
+	}
+
+	if (osr) {
+		/* osr->val2 should be unused */
+		switch (osr->val1) {
+		case 4:
+			osr_bits = BMI270_GYR_BWP_OSR4;
+			break;
+		case 2:
+			osr_bits = BMI270_GYR_BWP_OSR2;
+			break;
+		default:
+			osr_bits = BMI270_GYR_BWP_NORM;
+			break;
+		}
+
+		gyr_conf = BMI270_SET_BITS(gyr_conf, BMI270_GYR_BWP,
+					   osr_bits);
+	}
+
+	if (odr || osr) {
+		ret = reg_write(BMI270_REG_GYR_CONF, &gyr_conf, 1, dev);
+		if (ret != 0) {
+			return ret;
+		}
+
+		/* Assuming we have advance power save enabled */
+		k_usleep(BMI270_TRANSC_DELAY_SUSPEND);
+
+		pwr_ctrl &= BMI270_PWR_CTRL_MSK;
+		ret = reg_write(BMI270_REG_PWR_CTRL, &pwr_ctrl, 1, dev);
+		if (ret != 0) {
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static int set_gyro_range(const struct sensor_value *range,
+			  struct bmi270_data *dev)
+{
+	int ret = 0;
+	uint8_t gyr_range, reg;
+
+	ret = reg_read(BMI270_REG_GYR_RANGE, &gyr_range, 1, dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* range->val2 is unused */
+	switch (range->val1) {
+	case 125:
+		reg = BMI270_GYR_RANGE_125DPS;
+		dev->gyr_range = 125;
+		break;
+	case 250:
+		reg = BMI270_GYR_RANGE_250DPS;
+		dev->gyr_range = 250;
+		break;
+	case 500:
+		reg = BMI270_GYR_RANGE_500DPS;
+		dev->gyr_range = 500;
+		break;
+	case 1000:
+		reg = BMI270_GYR_RANGE_1000DPS;
+		dev->gyr_range = 1000;
+		break;
+	case 2000:
+		reg = BMI270_GYR_RANGE_2000DPS;
+		dev->gyr_range = 2000;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	gyr_range = BMI270_SET_BITS_POS_0(gyr_range, BMI270_GYR_RANGE, reg);
+
+	ret = reg_write(BMI270_REG_GYR_RANGE, &gyr_range, 1, dev);
+
+	return ret;
+}
+
+static int8_t write_config_file(struct bmi270_data *dev)
+{
+	int8_t ret = 0;
+	uint16_t index = 0;
+	uint8_t addr_array[2] = { 0 };
+
+	/* Disable loading of the configuration */
+	for (index = 0; index < sizeof(bmi270_config_file);
+	     index += BMI270_WR_LEN) {
+		/* Store 0 to 3 bits of address in first byte */
+		addr_array[0] = (uint8_t)((index / 2) & 0x0F);
+
+		/* Store 4 to 11 bits of address in the second byte */
+		addr_array[1] = (uint8_t)((index / 2) >> 4);
+
+		ret = reg_write(BMI270_REG_INIT_ADDR_0, addr_array, 2, dev);
+
+		k_usleep(450);
+
+		if (ret == 0) {
+			ret = reg_write(BMI270_REG_INIT_DATA,
+					(bmi270_config_file + index),
+					BMI270_WR_LEN, dev);
+			k_usleep(450);
+		}
+	}
+
+	return ret;
+}
+
+int bmi270_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct bmi270_data *drv_dev = dev->data;
+	uint8_t data[12];
+	int ret;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	ret = reg_read(BMI270_REG_ACC_X_LSB, data, 12, drv_dev);
```

Comment:
Because `_channel_get` supports individual channels, fetch should probably do the same. If someone is just interested in a particular sensor channel or group the driver will fetch more data than needed.

Reply:
@gmarull , this is not recommended as providing the options of fetching only 1 channel at a time will lead to data misalignment, in effect, you can have one channel from one measurement and one channel from another. It is hence forced here to fetch all channels together and then read which channels might be required.

Reply:
Agreed, this is the case for NXP sensors too.